linera-indexer/src/operations.rs

```diff
@@ -0,0 +1,234 @@
+                    operation,
+                    previous_operation,
+                    height,
+                    operation_index,
+                    hash: false,
+                };
+                info!(
+                    "register operation for {:?}:\n{:?}",
+                    chain_id, operation_view
+                );
+                plugin.operations.insert(&key, operation_view.clone())?;
+                let count = *plugin.count.get_mut(&chain_id).await?.unwrap_or(&mut 0);
+                plugin.count.insert(&chain_id, count + 1)?;
+                plugin.last.insert(&chain_id, (key, height, index))?;
+                plugin.save().await.map_err(IndexerError::ViewError)
+            }
+        }
+    }
+
+    /// Main function of plugin: registers the operations for a hashed value
+    pub async fn register(&self, value: &HashedValue) -> Result<(), IndexerError> {
+        let block = &value.inner().executed_block().block;
+        let chain_id = value.inner().chain_id();
+        for (i, op) in block.operations.iter().enumerate() {
+            match self
+                .register_operation(op.clone(), value.hash(), block.height, i, chain_id)
+                .await
+            {
+                Err(e) => return Err(e),
+                Ok(()) => continue,
+            }
+        }
+        Ok(())
+    }
+
+    /// Loads the plugin view from context
+    pub async fn load(context: C) -> Result<Self, IndexerError> {
+        let plugin = OperationsPluginInternal::load(context).await?;
+        Ok(OperationsPlugin(Arc::new(Mutex::new(plugin))))
+    }
+
+    pub fn schema(self) -> Schema<Self, EmptyMutation, EmptySubscription> {
+        Schema::build(self, EmptyMutation, EmptySubscription).finish()
+    }
+}
+
+fn hashed_operation(operation: ChainOperation<bool>) -> ChainOperation<CryptoHash> {
+    let hash = CryptoHash::new(&operation.operation);
+    ChainOperation {
+        operation: operation.operation,
+        height: operation.height,
+        previous_operation: operation.previous_operation,
+        operation_index: operation.operation_index,
+        hash,
+    }
+}
+
+#[derive(OneofObject)]
+pub enum OperationKeyKind {
+    Hash(CryptoHash),
+    Last(ChainId),
+}
+
+#[Object(name = "OperationsRoot")]
+impl<C> OperationsPlugin<C>
+where
+    C: Context + Send + Sync + 'static + Clone,
+    ViewError: From<C::Error>,
+{
+    /// Gets the operation associated to its hash
+    pub async fn operation(
+        &self,
+        key: OperationKeyKind,
+    ) -> Result<Option<ChainOperation<CryptoHash>>, IndexerError> {
+        let key = match key {
+            OperationKeyKind::Hash(hash) => hash,
+            OperationKeyKind::Last(chain_id) => {
+                match self.0.lock().await.last.get(&chain_id).await? {
+                    None => return Ok(None),
+                    Some((hash, _, _)) => hash,
+                }
+            }
+        };
```

Comment:
Why call the returned variables a `key` when it is actually a `hash`?
This is very minor point though.